File: aat-android/src/main/java/ch/bailu/aat/views/preferences/SolidDirectoryView.java
package ch.bailu.aat.views.preferences;

import ch.bailu.aat.preferences.SolidFile;
import ch.bailu.aat.util.ui.UiTheme;

public class SolidDirectoryView extends AbsSolidView {
    protected final SolidFile solid;



    public SolidDirectoryView(SolidFile s, UiTheme theme) {
        super(s, theme);
        solid = s;
    }


    @Override
    public void onRequestNewValue() {
        new SolidStringDialog(solid);
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/preferences/SolidDirectoryView.kt
package ch.bailu.aat.views.preferences

import android.content.Context
import ch.bailu.aat.util.ui.theme.UiTheme
import ch.bailu.aat.views.preferences.dialog.SolidStringDialog
import ch.bailu.aat_lib.preferences.SolidFile

open class SolidDirectoryView(
    private val context: Context,
    protected val solid: SolidFile,
    theme: UiTheme
) : AbsSolidView(
    context, solid, theme
) {
    override fun onRequestNewValue() {
        SolidStringDialog(context, solid)
    }
}
